### PR TITLE
Resolved asynchronous build gaps issues in khalti_flutter package and fixed lint issues in khalti package.

### PR DESCRIPTION
**Pull request by Biplab-Dutta**

### CODE REVIEW

File: packages/khalti_flutter/lib/src/page/bank_payment_page.dart [hunk 1]
```diff
@@ -156,22 +161,23 @@ class _BankTile extends StatelessWidget {
...
                       productName: config.productName,
-                      paymentType: paymentType,
+                      paymentType: widget.paymentType,
                       productUrl: config.productUrl,
                       additionalData: config.additionalData,
                       returnUrl: config.returnUrl,
                     );
                     await urlLauncher.launch(url);
+                    if (!mounted) return;
                     Navigator.pop(context);
```
Comment (sarbagyastha) [inline packages/khalti_flutter/lib/src/page/bank_payment_page.dart:181]: Let's do like this instead.

```dart
Builder(
  builder: (context) {
    final navigator = Navigator.of(context);
    
    Builder(
      builder: (ctx) async {
        navigator.pop();
     }
   )
  }
)

```

**Review by sarbagyastha — CHANGES_REQUESTED** [review]

File: packages/khalti_flutter/lib/src/page/bank_payment_page.dart (hunk 1 above)
Comment (Biplab-Dutta) [reply packages/khalti_flutter/lib/src/page/bank_payment_page.dart:181]: Should I do like this only in this file or in every file where I had made changes?

File: packages/khalti_flutter/lib/src/widget/khalti_progress_indicator.dart [hunk 2]
```diff
@@ -16,12 +16,13 @@ class KhaltiProgressIndicator extends StatefulWidget {
   final double size;
 
   @override
-  _KhaltiProgressIndicatorState createState() {
-    return _KhaltiProgressIndicatorState();
+  KhaltiProgressIndicatorState createState() {
```
Comment (sarbagyastha) [inline packages/khalti_flutter/lib/src/widget/khalti_progress_indicator.dart:19]: Keep the state private. You can update the return type to `State<KhaltiProgressIndicator>` to get rid of the lint issue.

**Review by sarbagyastha — CHANGES_REQUESTED** [review]

**Review by sarbagyastha — APPROVED** [review]
LGTM

**Pull request merged**